addons/account/models/ir_actions_report.py

```diff
@@ -1,16 +1,87 @@
 # -*- coding: utf-8 -*-
 
-from odoo import models, api, _
+import io
+
+from PIL import Image
+from PyPDF2 import PdfFileWriter, PdfFileReader
+from reportlab.lib import colors
+from reportlab.lib.units import cm
+from reportlab.lib.utils import ImageReader
+from reportlab.pdfgen import canvas
+
+from odoo import models, _
 from odoo.exceptions import UserError
+from odoo.tools import file_open
+
 
 class IrActionsReport(models.Model):
     _inherit = 'ir.actions.report'
 
+    def _add_banner_with_reference(self, record):
```

Comment:
That looks very very specific in terms of API, with both the image and the text contents hardcoded. Wouldn't it make more sense technically to have one or both be parameters?

Also, this should be a helper function, it doesn't use `self` and looks like it belongs in `odoo.tools.pdf` rather than `ir.actions.report`, no?

Reply:
So
```python
# odoo.tools.pdf
def add_banner(text=None, image=None, thickness=2*cm):
    """Add a banner on a PDF.

    :param text (str): The text to be displayed
    :param image (bytes): The image to be displayed
    :param thickness (float): The thickness of the banner in pixels
    """
```

As well as some parameters for the position of the logo and text? This might be overkill.